### PR TITLE
Fix for callback IOTHUB_CLIENT_CONNECTION_NO_NETWORK in AMQP

### PR DESCRIPTION
**Pull request by bikamani**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: iothub_client/src/iothubtransport_amqp_connection.c [hunk 1]
```diff
@@ -146,17 +146,17 @@ static void on_connection_state_changed(void* context, CONNECTION_STATE new_conn
 
 static void on_cbs_open_complete(void* context, CBS_OPEN_COMPLETE_RESULT open_complete_result)
 {
-    (void)context;
     (void)open_complete_result;
     if (open_complete_result != CBS_OPEN_OK)
     {
         LogError("CBS open failed");
+		update_state((AMQP_CONNECTION_INSTANCE*)context, AMQP_CONNECTION_STATE_ERROR);
```
Comment (ewertons) [inline iothub_client/src/iothubtransport_amqp_connection.c]: Please keep the indentation, and use spaces instead of tabs.

File: iothub_client/src/iothubtransport_amqp_connection.c [hunk 2]
```diff
@@ -146,17 +146,17 @@ static void on_connection_state_changed(void* context, CONNECTION_STATE new_conn
...
     }
 }
 
 static void on_cbs_error(void* context)
 {
-    (void)context;
+	(void)context;
```
Comment (ewertons) [inline iothub_client/src/iothubtransport_amqp_connection.c]: please revert this change, it's not needed.

File: iothub_client/src/iothubtransport_amqp_connection.c [hunk 3]
```diff
@@ -271,7 +271,7 @@ static int create_cbs_handle(AMQP_CONNECTION_INSTANCE* instance)
         LogError("Failed to create the CBS connection.");
     }
     // Codes_SRS_IOTHUBTRANSPORT_AMQP_CONNECTION_09_031: [`instance->cbs_handle` shall be opened using `cbs_open_async`]
-    else if (cbs_open_async(instance->cbs_handle, on_cbs_open_complete, instance->cbs_handle, on_cbs_error, instance->cbs_handle) != RESULT_OK)
+    else if (cbs_open_async(instance->cbs_handle, on_cbs_open_complete, instance, on_cbs_error, instance) != RESULT_OK)
```
Comment (ewertons) [inline iothub_client/src/iothubtransport_amqp_connection.c]: Thanks!

File: iothub_client/src/iothubtransport_amqp_connection.c (hunk 1 above)
Comment (bikamani) [reply iothub_client/src/iothubtransport_amqp_connection.c]: I am using tabs and indentation looks fine in my editor, not sure why it looks off here. I will double check

File: iothub_client/src/iothubtransport_amqp_connection.c (hunk 2 above)
Comment (bikamani) [reply iothub_client/src/iothubtransport_amqp_connection.c]: Same here, no change and using tab.
